File: ThunderCloud/UIWindow+TSCWindow.m
//
//  UIWindow+TSCWindow.m
//  ThunderCloud
//
//  Created by Simon Mitchell on 17/09/2014.
//  Copyright (c) 2014 threesidedcube. All rights reserved.
//

#import "UIWindow+TSCWindow.h"
#import <ThunderCloud/ThunderCloud-Swift.h>

@implementation UIWindow (TSCWindow)

- (void)motionEnded:(UIEventSubtype)motion withEvent:(UIEvent *)event
{
    if ([[NSBundle mainBundle] infoDictionary][@"TSCStormLoginDisabled"]) {
        return;
    }
    
    if (event.type == UIEventSubtypeMotionShake) {
        
        // Attempt to disable editing localisations for app store releases.
        
#ifdef DEBUG
        
        [[TSCLocalisationController sharedController] toggleEditing];

#else
        NSString *provisionPath = [[NSBundle mainBundle] pathForResource:@"embedded" ofType:@"mobileprovision"];
        if (provisionPath) {
            [[TSCLocalisationController sharedController] toggleEditing];
        }
#endif
    }
}

@end

File: ThunderCloudSwift/Localisations/UIWindow+TSCWindow.swift
//
//  UIWindow+TSCWindow.swift
//  ThunderCloud
//
//  Created by Simon Mitchell on 17/09/2014.
//  Copyright (c) 2014 threesidedcube. All rights reserved.
//

import UIKit

/**
 An extension on `UIWindow` that enables editing of localizations
 by responding to a system-wide shake gesture.
 */
extension UIWindow {
    open override func motionEnded(_ motion: UIEvent.EventSubtype, with event: UIEvent?) {
        guard
            !(Bundle.main.infoDictionary?["TSCStormLoginDisabled"] as? Bool ?? true),
            motion == .motionShake
        else { return }

        // Attempt to disable editing localizations for app store releases.
        #if DEBUG
        LocalisationController.shared.toggleEditing()
        #else
        if Bundle.main.path(forResource: "embedded", ofType: "mobileprovision") != nil {
            LocalisationController.shared().toggleEditing()
        }
        #endif
    }
}
